dist/dashboard-utils.js

```diff
@@ -1490,117 +1490,208 @@ angular.module('enplug.utils').factory('EndpointOptions', ['$log', 'Environment'
 angular.module('enplug.utils.environment', []).provider('Environment', function () {
     'use strict';
 
-    var cookieName = 'ENVIRONMENT';
-
-    // Environment list
-    this.PRODUCTION = 'prod';
-    this.STAGING = 'staging';
-    this.DEV = 'dev';
+    // TODO remove console.log statements
+    var cookieName = 'ENVIRONMENT',
+        paramName = cookieName.toLowerCase(),
+        domainEnvMap,
+        fallbackEnv,
+        currentEnv;
+
+    // immutable Environment list on 'this'
+    Object.defineProperties( this, {
+        PRODUCTION: {
+            configurable: false,
+            enumerable: true,
+            writeable: false,
+            value: 'prod'
+        },
+        STAGING: {
+            configurable: false,
+            enumerable: true,
+            writeable: false,
+            value: 'staging'
+        },
+        DEV: {
+            configurable: false,
+            enumerable: true,
+            writeable: false,
+            value: 'dev'
+        }
+    });
+
+    // set 'fallback' environment enum here for easy changing later
+    fallbackEnv = this.STAGING;
+
+    // maps domains to expected environment
+    // todo remove these notes below (maybe?)
+    //  prod dashboard is on .com
+    //  prod apps are on .net
+    //  staging apps and dashboard are on .in
+    //  local is for local dev envs (must be set in your computers "hosts" file to 127.0.0.1)
+    domainEnvMap = Object.assign( Object.create( null ), {
```

Comment:
I happened to still be watching this repository so I saw this PR (I'm unsubscribing now), and as I was skimming through it the call to `Object.assign()` stood out. I don't think this is supported in any versions of IE: [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign).

Best of luck!